tests/test_telemetry.py

```diff
@@ -371,6 +370,62 @@ def test_app_dependencies_loaded(self):
             if not seen:
                 raise Exception(dependency + " not recieved in app-dependencies-loaded message")
 
+    def setup_app_dependency_loaded_not_sent_dependency_collection_disabled(self):
+        weblog.get("/enable_product")
+
+    @missing_feature(
+        context.library in ("dotnet", "nodejs", "java", "golang", "python"),
+        reason="Weblog endpoint GET /enable_product is not implemented yet. ",
+    )
+    def test_app_started_product_info(self):
+        """Assert that product information is accurately reported by telemetry"""
+
+        def validator(data):
+            if data["request"]["content"].get("request_type") == "app-started":
+                content = data["request"]["content"]
+                products = content["payload"]["products"]
+                assert (
+                    "appsec" in products
+                ), "Product information is not accurately reported by telemetry on app-started event"
+
+        self.validate_library_telemetry_data(validator)
+
+    @irrelevant(library="php")
+    @irrelevant(library="cpp")
+    @irrelevant(library="golang")
+    @irrelevant(library="ruby")
+    @bug(
+        library="python",
+        reason="""
+            configuration is not properly populating for python
+        """,
+    )
+    def test_app_started_client_configuration(self):
+        """Assert that default and other configurations that are applied upon start time are sent with the app-started event"""
+        test_configuration = {
+            "dotnet": {},
+            "nodejs": {"hostname", "port", "appsec.enabled"},
+            # to-do :need to add configuration keys once python bug is fixed
+            "python": {},
+            "java": {"trace.agent.port", "telemetry.heartbeat.interval"},
+        }
+        configurationMap = test_configuration[context.library.library]
+
+        def validator(data):
+            if data["request"]["content"].get("request_type") == "app-started":
+                content = data["request"]["content"]
+                cnt = 0
+                configurations = content["payload"]["configuration"]
+                for cnf in configurations:
+                    if cnf["name"] in configurationMap:
+                        cnt += 1
+                if cnt != len(list(configurationMap)):
+                    raise Exception(
+                        "Client Configuration information is not accurately reported by telemetry on app-started event"
+                    )
```

Comment:
can you assert that the cnf["name"] in the map so that the error message can display exactly which config is missing or has an incorrect value